### PR TITLE
[Linux] Fix device connection handler in CHIPBluezHelper

### LINKED ISSUE #3548: Chip tool server crash on raspi4 when doing connect using python chip-device-ctrl
When trying to connect to the chip device on raspi4 

Stack trace from the python device controller
-------------------------------------------------------
python3 chip-device-ctrl
Chip Device Controller Shell

chip-device-ctrl > help

Available commands (type help <name> for more information):
============================================================
ble-adapter-print   ble-disconnect    close                      
ble-adapter-select  ble-scan          connect                    
ble-connect         ble-scan-connect  set-pairing-wifi-credential
ble-debug-log       btp-connect     

chip-device-ctrl > ble-scan-connect CHIP-2976
2020-10-30 11:07:01,030 ChipBLEMgr   INFO     BLE is ready!
2020-10-30 11:07:01,432 ChipBLEMgr   INFO     adding to scan list:
2020-10-30 11:07:01,432 ChipBLEMgr   INFO     
2020-10-30 11:07:01,432 ChipBLEMgr   INFO     Name =    None                                                                            
2020-10-30 11:07:01,432 ChipBLEMgr   INFO     ID =      BF518E80-E89D-47FF-A645-4B2D1D9E27D2                                            
2020-10-30 11:07:01,432 ChipBLEMgr   INFO     RSSI =    -46                                                                             
2020-10-30 11:07:01,433 ChipBLEMgr   INFO     ADV data: {
    kCBAdvDataChannel = 37;
    kCBAdvDataIsConnectable = 1;
    kCBAdvDataServiceData =     {
        FEAF = {length = 3, bytes = 0x000f00};
    };
}
2020-10-30 11:07:01,433 ChipBLEMgr   INFO     
2020-10-30 11:07:05,655 ChipBLEMgr   INFO     adding to scan list:
2020-10-30 11:07:05,655 ChipBLEMgr   INFO     
2020-10-30 11:07:05,655 ChipBLEMgr   INFO     Name =    None                                                                            
2020-10-30 11:07:05,655 ChipBLEMgr   INFO     ID =      C721EBDD-B7EE-4AE0-949B-49481DC4CB2C                                            
2020-10-30 11:07:05,655 ChipBLEMgr   INFO     RSSI =    -56                                                                             
2020-10-30 11:07:05,656 ChipBLEMgr   INFO     ADV data: {
    kCBAdvDataChannel = 37;
    kCBAdvDataIsConnectable = 1;
    kCBAdvDataServiceData =     {
        FEAF = {length = 17, bytes = 0x1001010101000100010000000000000000};
    };
    kCBAdvDataServiceUUIDs =     (
        FEAF
    );
}
2020-10-30 11:07:05,656 ChipBLEMgr   INFO     
2020-10-30 11:07:05,656 ChipBLEMgr   INFO     scanning stopped
2020-10-30 11:07:05,656 ChipBLEMgr   INFO     trying to connect to CHIP-2976
2020-10-30 11:07:08,453 ChipBLEMgr   INFO     Discovering services
2020-10-30 11:07:08,455 ChipBLEMgr   INFO     Discovering services
2020-10-30 11:07:08,615 ChipBLEMgr   INFO     connect success
chip-device-ctrl > set-pairing-wifi-credential testap testtest
WiFi credential set
chip-device-ctrl > connect 12345678
CHIP: [BLE] subscribe complete, ep = 0x114690c88
CHIP: [BLE] connect handshake timed out, closing ep 0x114690c88
CHIP: [BLE] Auto-closing end point's BLE connection.
2020-10-30 11:07:47,426 ChipBLEMgr   INFO     BLE disconnected, error = None

Here is stack trace from the raspi , chip tool server...
------------------------------------------------------
<suggested fix, suggested enhancement>
CHIP:DL: SET service data to {'0xFEAF': <[byte 0x10, 0x01, 0x01, 0x01, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00]>}
CHIP:DL: HandlePlatformSpecificBLEEvent , 16388
CHIP:DL: CHIPoBLE advertising config complete
CHIP:DL: RegisterAdvertisement complete
CHIP:DL: HandlePlatformSpecificBLEEvent , 16389
[Thread 0xfffff56cefc0 (LWP 2440211) exited]

Thread 4 "chip-tool-serve" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0xfffff5ecffc0 (LWP 2440210)]
0x0000fffff7b752b0 in g_str_hash () from /lib/aarch64-linux-gnu/libglib-2.0.so.0
(gdb) bt
#0  0x0000fffff7b752b0 in g_str_hash () at /lib/aarch64-linux-gnu/libglib-2.0.so.0
#1  0x0000fffff7b73a4c in g_hash_table_lookup () at /lib/aarch64-linux-gnu/libglib-2.0.so.0
#2  0x0000aaaaaaae634c in chip::DeviceLayer::Internal::GetBluezConnectionViaDevice(chip::DeviceLayer::Internal::BluezEndpoint*)
    (apEndpoint=0xaaaaaacd5f00) at ../../third_party/connectedhomeip/src/platform/Linux/CHIPBluezHelper.cpp:1115
#3  0x0000aaaaaaae4430 in chip::DeviceLayer::Internal::BluezCharacteristicAcquireWrite(BluezGattCharacteristic1*, GDBusMethodInvocation*, GVariant*, gpointer) (aChar=0xffffe4003b50, aInvocation=0xffffe802c480, aOptions=0xffffe8027520, apEndpoint=0xaaaaaacd5f00)
    at ../../third_party/connectedhomeip/src/platform/Linux/CHIPBluezHelper.cpp:407
#4  0x0000fffff734d148 in  () at /lib/aarch64-linux-gnu/libffi.so.8
#5  0x0000fffff734c748 in  () at /lib/aarch64-linux-gnu/libffi.so.8
#6  0x0000fffff7c8da50 in g_cclosure_marshal_generic () at /lib/aarch64-linux-gnu/libgobject-2.0.so.0
#7  0x0000fffff7c8d058 in g_closure_invoke () at /lib/aarch64-linux-gnu/libgobject-2.0.so.0
#8  0x0000fffff7ca1c74 in  () at /lib/aarch64-linux-gnu/libgobject-2.0.so.0
#9  0x0000aaaaaab6cfec in _bluez_gatt_characteristic1_skeleton_handle_method_call
    (connection=0xaaaaaacc3040, sender=0xffffe802b580 ":1.5", object_path=0xffffe802d210 "/chipoble/3c0d/service/c1", interface_name=0xffffe802c090 "org.bluez.GattCharacteristic1", method_name=0xffffe802c070 "AcquireWrite", parameters=0xffffe8004af0, invocation=0xffffe802c480, user_data=0xffffe4003b50) at gen/platform/Linux/dbus/bluez/DbusBluez.c:12258
#10 0x0000fffff7e210e8 in  () at /lib/aarch64-linux-gnu/libgio-2.0.so.0
#11 0x0000fffff7e05704 in  () at /lib/aarch64-linux-gnu/libgio-2.0.so.0
#12 0x0000fffff7b879f0 in g_main_context_dispatch () at /lib/aarch64-linux-gnu/libglib-2.0.so.0
#13 0x0000fffff7b87d98 in  () at /lib/aarch64-linux-gnu/libglib-2.0.so.0
#14 0x0000fffff7b880f0 in g_main_loop_run () at /lib/aarch64-linux-gnu/libglib-2.0.so.0
#15 0x0000aaaaaaaec528 in chip::DeviceLayer::GDBus_Thread() ()
    at ../../third_party/connectedhomeip/src/platform/Linux/PlatformManagerImpl.cpp:45
#16 0x0000aaaaaaaee9a4 in std::__invoke_impl<void, void (*)()>(std::__invoke_other, void (*&&)())
    (__f=@0xffffe8006cd8: 0xaaaaaaaec508 <chip::DeviceLayer::GDBus_Thread()>) at /usr/include/c++/10/bits/invoke.h:60
#17 0x0000aaaaaaaee930 in std::__invoke<void (*)()>(void (*&&)()) (__fn=@0xffffe8006cd8: 0xaaaaaaaec508 <chip::DeviceLayer::GDBus_Thread()>)
    at /usr/include/c++/10/bits/invoke.h:95
#18 0x0000aaaaaaaee8cc in std::thread::_Invoker<std::tuple<void (*)()> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) (this=0xffffe8006cd8)
    at /usr/include/c++/10/thread:264
#19 0x0000aaaaaaaee8a0 in std::thread::_Invoker<std::tuple<void (*)()> >::operator()() (this=0xffffe8006cd8)
    at /usr/include/c++/10/thread:271
#20 0x0000aaaaaaaee880 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)()> > >::_M_run() (this=0xffffe8006cd0)
    at /usr/include/c++/10/thread:215
--Type <RET> for more, q to quit, c to continue without paging--
#21 0x0000fffff7787e1c in  () at /lib/aarch64-linux-gnu/libstdc++.so.6
#22 0x0000fffff7f8ef74 in start_thread (arg=0x0) at pthread_create.c:463
#23 0x0000fffff76183dc in thread_start () at ../sysdeps/unix/sysv/linux/aarch64/clone.S:78CHIP:DL: SET service data to {'0xFEAF': <[byte 0x10, 0x01, 0x01, 0x01, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00]>}
CHIP:DL: HandlePlatformSpecificBLEEvent , 16388
CHIP:DL: CHIPoBLE advertising config complete
CHIP:DL: RegisterAdvertisement complete
CHIP:DL: HandlePlatformSpecificBLEEvent , 16389
[Thread 0xfffff56cefc0 (LWP 2440211) exited]

Thread 4 "chip-tool-serve" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0xfffff5ecffc0 (LWP 2440210)]
0x0000fffff7b752b0 in g_str_hash () from /lib/aarch64-linux-gnu/libglib-2.0.so.0
(gdb) bt
#0  0x0000fffff7b752b0 in g_str_hash () at /lib/aarch64-linux-gnu/libglib-2.0.so.0
#1  0x0000fffff7b73a4c in g_hash_table_lookup () at /lib/aarch64-linux-gnu/libglib-2.0.so.0
#2  0x0000aaaaaaae634c in chip::DeviceLayer::Internal::GetBluezConnectionViaDevice(chip::DeviceLayer::Internal::BluezEndpoint*)
    (apEndpoint=0xaaaaaacd5f00) at ../../third_party/connectedhomeip/src/platform/Linux/CHIPBluezHelper.cpp:1115
#3  0x0000aaaaaaae4430 in chip::DeviceLayer::Internal::BluezCharacteristicAcquireWrite(BluezGattCharacteristic1*, GDBusMethodInvocation*, GVariant*, gpointer) (aChar=0xffffe4003b50, aInvocation=0xffffe802c480, aOptions=0xffffe8027520, apEndpoint=0xaaaaaacd5f00)
    at ../../third_party/connectedhomeip/src/platform/Linux/CHIPBluezHelper.cpp:407
#4  0x0000fffff734d148 in  () at /lib/aarch64-linux-gnu/libffi.so.8
#5  0x0000fffff734c748 in  () at /lib/aarch64-linux-gnu/libffi.so.8
#6  0x0000fffff7c8da50 in g_cclosure_marshal_generic () at /lib/aarch64-linux-gnu/libgobject-2.0.so.0
#7  0x0000fffff7c8d058 in g_closure_invoke () at /lib/aarch64-linux-gnu/libgobject-2.0.so.0
#8  0x0000fffff7ca1c74 in  () at /lib/aarch64-linux-gnu/libgobject-2.0.so.0
#9  0x0000aaaaaab6cfec in _bluez_gatt_characteristic1_skeleton_handle_method_call
    (connection=0xaaaaaacc3040, sender=0xffffe802b580 ":1.5", object_path=0xffffe802d210 "/chipoble/3c0d/service/c1", interface_name=0xffffe802c090 "org.bluez.GattCharacteristic1", method_name=0xffffe802c070 "AcquireWrite", parameters=0xffffe8004af0, invocation=0xffffe802c480, user_data=0xffffe4003b50) at gen/platform/Linux/dbus/bluez/DbusBluez.c:12258
#10 0x0000fffff7e210e8 in  () at /lib/aarch64-linux-gnu/libgio-2.0.so.0
#11 0x0000fffff7e05704 in  () at /lib/aarch64-linux-gnu/libgio-2.0.so.0
#12 0x0000fffff7b879f0 in g_main_context_dispatch () at /lib/aarch64-linux-gnu/libglib-2.0.so.0
#13 0x0000fffff7b87d98 in  () at /lib/aarch64-linux-gnu/libglib-2.0.so.0
#14 0x0000fffff7b880f0 in g_main_loop_run () at /lib/aarch64-linux-gnu/libglib-2.0.so.0
#15 0x0000aaaaaaaec528 in chip::DeviceLayer::GDBus_Thread() ()
    at ../../third_party/connectedhomeip/src/platform/Linux/PlatformManagerImpl.cpp:45
#16 0x0000aaaaaaaee9a4 in std::__invoke_impl<void, void (*)()>(std::__invoke_other, void (*&&)())
    (__f=@0xffffe8006cd8: 0xaaaaaaaec508 <chip::DeviceLayer::GDBus_Thread()>) at /usr/include/c++/10/bits/invoke.h:60
#17 0x0000aaaaaaaee930 in std::__invoke<void (*)()>(void (*&&)()) (__fn=@0xffffe8006cd8: 0xaaaaaaaec508 <chip::DeviceLayer::GDBus_Thread()>)
    at /usr/include/c++/10/bits/invoke.h:95
#18 0x0000aaaaaaaee8cc in std::thread::_Invoker<std::tuple<void (*)()> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) (this=0xffffe8006cd8)
    at /usr/include/c++/10/thread:264
#19 0x0000aaaaaaaee8a0 in std::thread::_Invoker<std::tuple<void (*)()> >::operator()() (this=0xffffe8006cd8)
    at /usr/include/c++/10/thread:271
#20 0x0000aaaaaaaee880 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)()> > >::_M_run() (this=0xffffe8006cd0)
    at /usr/include/c++/10/thread:215
--Type <RET> for more, q to quit, c to continue without paging--
#21 0x0000fffff7787e1c in  () at /lib/aarch64-linux-gnu/libstdc++.so.6
#22 0x0000fffff7f8ef74 in start_thread (arg=0x0) at pthread_create.c:463
#23 0x0000fffff76183dc in thread_start () at ../sysdeps/unix/sysv/linux/aarch64/clone.S:78

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem
Chip tool server crash on raspi4 when doing connect using python chip-device-ctrl

 #### Summary of Changes
Fix device connection handle in CHIPBluezHelper

We need to handle device connection both this function and BluezSignalInterfacePropertiesChanged
When a device is connected for first time, this function will be triggerred.
The future connections for the same device will trigger ``Connect'' property change.

 #### Test
Tested using chip-device-controller python script.

 Fixes #3548

### CODE REVIEW

File: src/platform/Linux/CHIPBluezHelper.cpp [hunk 1]
```diff
@@ -997,7 +1001,9 @@ static void BluezHandleNewDevice(BluezDevice1 * device, BluezEndpoint * apEndpoi
         conn->mpDevice      = static_cast<BluezDevice1 *>(g_object_ref(device));
         conn->mpEndpoint    = apEndpoint;
         BluezConnectionInit(conn);
-
+        apEndpoint->mpPeerDevicePath = g_strdup(g_dbus_proxy_get_object_path(G_DBUS_PROXY(device)));
+        ChipLogDetail(DeviceLayer, "coonnected, insert, conn:%p, c1:%p, c2:%p and %s", conn, conn->mpC1, conn->mpC2,
```
Comment (mspang) [inline src/platform/Linux/CHIPBluezHelper.cpp:1005]: typo: connectected -> Connected

Can we avoid logging pointers?

File: src/platform/Linux/CHIPBluezHelper.cpp [hunk 2]
```diff
@@ -921,8 +921,8 @@ static void BluezSignalInterfacePropertiesChanged(GDBusObjectManagerClient * aMa
                             conn->mpEndpoint    = endpoint;
                             BluezConnectionInit(conn);
                             endpoint->mpPeerDevicePath = g_strdup(g_dbus_proxy_get_object_path(aInterface));
-                            ChipLogError(DeviceLayer, "coonnected, insert, conn:%p, c1:%p, c2:%p and %s", conn, conn->mpC1,
-                                         conn->mpC2, endpoint->mpPeerDevicePath);
+                            ChipLogDetail(DeviceLayer, "coonnected, insert, conn:%p, c1:%p, c2:%p and %s", conn, conn->mpC1,
```
Comment (BroderickCarlin) [inline src/platform/Linux/CHIPBluezHelper.cpp:924]: nit: `coonnected`

Comment (erjiaqing) [reply src/platform/Linux/CHIPBluezHelper.cpp]: Thanks!

File: src/platform/Linux/CHIPBluezHelper.cpp (hunk 1 above)
Comment (erjiaqing) [reply src/platform/Linux/CHIPBluezHelper.cpp]: Just logging address and peer path now.

**Review by mspang — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

**Review by rwalker-apple — APPROVED** [review]

**Pull request merged**